app/api/items/items.go

```diff
@@ -45,6 +45,7 @@ func (srv *Service) SetRoutes(router chi.Router) {
 	router.Get("/items/{item_id}/entry-state",
 		service.AppHandler(srv.getEntryState).ServeHTTP)
 	router.Post("/attempts/{attempt_id}/items/{item_id}/enter", service.AppHandler(srv.enter).ServeHTTP)
+	router.Delete("/attempts/{attempt_id}", service.AppHandler(srv.endAttempt).ServeHTTP)
```

Comment:
It is not really deleting attempt, as we may allow to visit again the attempt but in read-only and allow teach to grade it. I think it would be misleading to use a delete here. I prefer to keep "POST actions" as others, so something like `POST /attempts/{attempt_id}/end` (or `leave`)

Reply:
fixed